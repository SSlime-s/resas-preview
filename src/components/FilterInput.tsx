import { styled } from "@kuma-ui/core";
import { MdSearch } from "react-icons/md";

interface Props extends React.InputHTMLAttributes<HTMLInputElement> {}
export function FilterInput({ ...props }: Props) {
	return (
		<Wrap>
			<MdSearch size={18} title="検索" />
			<Input type="search" {...props} />
		</Wrap>
	);
}

const Wrap = styled.label`
	display: grid;
	grid-template-columns: max-content 1fr;
	align-items: center;

	gap: 0.25rem;
	padding: 0.25rem;
	padding-left: 0.5rem;

	cursor: default;

	background: var(--background);

	border-radius: var(--radius-sm);

	&:focus-within {
		outline: auto;
		outline: auto -webkit-focus-ring-color;
		box-shadow: 0 0 0 5px Canvas;
	}
`;

const Input = styled.input`
	border: none;
	background: none;
	color: inherit;
	font-size: 1rem;
	width: 100%;

	&:focus-visible {
		outline: none;
		box-shadow: none;
	}
`;
