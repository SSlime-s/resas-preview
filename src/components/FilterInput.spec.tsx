import { render } from "@testing-library/react";

import { FilterInput } from "./FilterInput";

function setup() {
	const result = render(<FilterInput />);

	return { ...result };
}

describe("Components/FilterInput", () => {
	it("スナップショット", () => {
		const { container } = setup();

		expect(container).toMatchSnapshot();
	});
});
