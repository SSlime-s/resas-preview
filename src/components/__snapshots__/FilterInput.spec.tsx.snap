// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Components/FilterInput > スナップショット 1`] = `
<div>
  <label
    class="🐻-92457733"
  >
    <svg
      fill="currentColor"
      height="18"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 24 24"
      width="18"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        検索
      </title>
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M15.5 14h-.79l-.28-.27A6.471 6.471 0 0 0 16 9.5 6.5 6.5 0 1 0 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
      />
    </svg>
    <input
      class="🐻-66788598"
      type="search"
    />
  </label>
</div>
`;
